### PR TITLE
Fix bridge error in vision

### PR DESCRIPTION
**Pull request by vivekandathil**
## Related Issue(s)
  - Issue #211 

## Changes Made
- Removed lines causing bridge error when running the vision simulator

## Tests Done
- Verify catkin build
- Verify with simulator

### CODE REVIEW

**Review by MrMondrian — DISMISSED** [review]

File: catkin_ws/src/vision/src/point_cloud.py
Comment (antoinedang) [inline catkin_ws/src/vision/src/point_cloud.py]: was this causing errors? I'm pretty sure this is important for the vision package, it uses this topic to get point cloud information. It should be replaced with listening directly to the point cloud topic and extracting rgb/xyz images directly but for now this should stay

**Review by antoinedang — CHANGES_REQUESTED** [review]

**Comment by MrMondrian** [comment]
This didn’t work on either of our machines and the code in point cloud .py is only intended for debugging

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by antoinedang** [comment]
> This didn’t work on either of our machines and the code in point cloud .py is only intended for debugging 

What didn't work about it? Cuz just deleting this line will break vision 100%. So I don't think we can merge this if it fixes one problem and causes another. Also I thought it was supposed to be a "real" file, I don't think having the point cloud math in object_detection_utils makes much sense. Or maybe we should rename it to like point_cloud_debug.py and have a point_cloud.py that handles the camera feeds and republishes a point cloud

**Comment by MrMondrian** [comment]
The python implementation of the ros point cloud stuff is too slow. We can't transport point cloud data via ros for the time being. I profiled the function and the point cloud math is very fast even for large images but generating a point cloud ros message takes many seconds

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by MrMondrian** [comment]
When I wrote this I kept the vision point cloud and debugging point cloud separate. Maybe someone changed that but that would be a bad idea imo

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

File: catkin_ws/src/vision/src/point_cloud.py
Comment (MrMondrian) [reply catkin_ws/src/vision/src/point_cloud.py]: the bug can be fixed by removing the optional "bgr8" argument

**Comment by antoinedang** [comment]
> When I wrote this I kept the vision point cloud and debugging point cloud separate. Maybe someone changed that but that would be a bad idea imo
> The python implementation of the ros point cloud stuff is too slow. We can't transport point cloud data via ros for the time being. I profiled the function and the point cloud math is very fast even for large images but generating a point cloud ros message takes many seconds

Ok makes sense, lets have object_detection_utils.py import point_cloud.py and use its functions, and then if we can have debugging be done by directly running the point_cloud.py

**Comment by MrMondrian** [comment]
@antoinedang waiting on your approval

File: catkin_ws/src/vision/src/object_detection_utils.py
Comment (antoinedang) [inline catkin_ws/src/vision/src/object_detection_utils.py]: should be self.updatePointCloud() I think

File: catkin_ws/src/vision/src/point_cloud.py
Comment (antoinedang) [inline catkin_ws/src/vision/src/point_cloud.py]: could you make the references to 'rbg' be 'rgb' (nitpick)

File: catkin_ws/src/vision/src/point_cloud.py
Comment (antoinedang) [inline catkin_ws/src/vision/src/point_cloud.py]: since depth is a global variable no need for it to be passed as an argument I think

File: catkin_ws/src/vision/src/object_detection_utils.py
Comment (antoinedang) [inline catkin_ws/src/vision/src/object_detection_utils.py]: I don't think this works here for 2 reasons: since we never actually do any front cam detection until a point cloud exists (is not None), we'll never get to the point in the code where getPointCloud() is called. It'll be a sort of deadlock state where object_detection.py is waiting for the State class to have a non-None point cloud attribute, but the State class only has a non-None point cloud attribute when it's getPointCloud() function is called, and its only called after object_detection.py proceeds with a detection. Also, the same problem would happen with the pausing: getPointCloud() should only be called after the State class is paused, which means that the point cloud will never actually update since it's always paused

Comment (antoinedang) [reply catkin_ws/src/vision/src/object_detection_utils.py]: I think it might be cleaner and avoid these problems by creating a subscriber inside the State class for image and camera_info topics and using the functions from point_cloud.py to turn them into a usable point cloud array. It would also avoid having to use global variables for depth

**Review by antoinedang — CHANGES_REQUESTED** [review]
good code and I like the depth_scaling_factor changing based on sim vs. real, just re-structuring would fix it

File: catkin_ws/src/vision/src/object_detection_utils.py
Comment (antoinedang) [inline catkin_ws/src/vision/src/object_detection_utils.py:45]: we don't need an RGB subscriber normally

File: catkin_ws/src/vision/src/object_detection_utils.py
Comment (antoinedang) [inline catkin_ws/src/vision/src/object_detection_utils.py]: lmao what does "Yo" mean?

**Review by antoinedang — DISMISSED** [review]
since its been a while and you guys will be fixing this stuff anyway, I'm good to merge but you'll have to make sure in the future that we can clean this up (just the rgb subscriber stuff and commenting, handling the value error ,etc.). I'll leave it up to you

**Comment by antoinedang** [comment]
Oh also make sure the integration test doesnt fail before you merge

File: catkin_ws/src/vision/src/object_detection_utils.py
Comment (vivekandathil) [reply catkin_ws/src/vision/src/object_detection_utils.py]: Ah sorry about that! Forgot to remove after testing haha

**Comment by antoinedang** [comment]
@vivekandathil good to merge for me, especially since it'll unblock everyone working on front cam stuff. Btw feel free to resolve any conversations you've addressed before, don't wanna do it for you in case you are still addressing any but yeah i dont mind if you do

**Comment by vivekandathil** [comment]
Sounds good, thanks! @antoinedang 
I'll create tasks for fixing the RGB subscriber & ValueError parts after merging

**Pull request merged**
